lib/chef/resource/windows_task.rb

```diff
@@ -94,25 +90,35 @@ def after_created
 
       private
 
+      # Validate the passed value is numeric values only if it is a string
+      def numeric_value_in_string?(val)
+        return false if val.is_a?(String) && /\D/ =~ val # \D is any non-numeric value
+        true
+      end
+
       def validate_random_delay(random_delay, frequency)
         if [:once, :on_logon, :onstart, :on_idle, :none].include? frequency
           raise ArgumentError, "`random_delay` property is supported only for frequency :minute, :hourly, :daily, :weekly and :monthly"
         end
 
-        raise ArgumentError, "Invalid value passed for `random_delay`. Please pass seconds as a String e.g. '60'." if random_delay.to_i == 0
+        raise ArgumentError, "Invalid value passed for `random_delay`. Please pass seconds as an Integer (e.g. 60) or a String with numeric values only (e.g. '60')." unless numeric_value_in_string?(random_delay)
       end
 
       def validate_start_day(start_day, frequency)
         if [:once, :on_logon, :onstart, :on_idle, :none].include? frequency
           raise ArgumentError, "`start_day` property is not supported with frequency: #{frequency}"
         end
+
+        # make sure the start_day is in MM/DD/YYYY format: http://rubular.com/r/cgjHemtWl5
+        raise ArgumentError, "`start_day` property must be in the MM/DD/YYYY format." unless /^(0[1-9]|1[012])[- \/.](0[1-9]|[12][0-9]|3[01])[- \/.](19|20)\d\d$/.match?(start_day)
```

Comment:
I believe this only works for US locale machines. The `start_day` property is a string, which I believe means the date would be passed to `schtasks.exe` wholesale on the command line.  On my UK english locale system this would cause schtasks.exe to fail with an error, as `schtasks.exe` follows the local system locale for command line input.  See the output on my system for an example:

```
C:\projects\other\windows_task_testing> SCHTASKS /Create /SC HOURLY /TN accountant /TR calc.exe /ST 12:00 /ET 14:00 /SD **13/12/2017** /ED 14/12/2017
SUCCESS: The scheduled task "accountant" has successfully been created.
C:\projects\other\windows_task_testing> SCHTASKS /Create /SC HOURLY /TN accountant-us /TR calc.exe /ST 12:00 /ET 14:00 /SD **12/13/2017** /ED 12/13/2017
ERROR: Incorrect Start Date.
```

As an aside, I see we don't have corresponding `end_date` and `end_time` properties (Task expiry time in Task Scheduler).

Reply:
I take it back, the existing docs do state to use MM/DD/YYYY irrespective of the system locale and any local knowledge of schtasks.exe.